internal/workload/wrapper.go

```diff
@@ -359,3 +364,27 @@ func getHostPorts(workload *v1.Pod) ([]int32, error) {
 	}
 	return hostPorts, nil
 }
+
+func isPortAvailable(port int32) bool {
+	conn, err := net.Listen("tcp", fmt.Sprintf("localhost:%d", port))
```

Comment:
How will it work in our e2e tests, where the agent runs in a container?

Reply:
We already do that, so should not be issue: https://github.com/project-flotta/flotta-operator/blob/066b8f6c2f527f01ff481b61b8fb8f519b0ff1be/test/e2e/e2e_test.go#L145

Reply:
but I guess now the test will fail because the pod is not created since the port is taken, so no systemd service, right?